src/app/compiler.js

```diff
@@ -119,7 +123,11 @@ function Compiler (editor, queryParams, handleGithubCall, updateFiles) {
     if (missingInputs !== undefined && missingInputs.length > 0) {
       compile(missingInputs);
     } else if (noFatalErrors) {
-      self.event.trigger('compilationFinished', [true, data, editor.getValue()]);
+      self.lastCompilationResult = {
+        data: data,
+        source: source
+      }
+      self.event.trigger('compilationFinished', [true, data, source]);
```

Comment:
Why? If there is an `errors` property, but it only contains warnings, then the event is triggered in line 115 and in this line.